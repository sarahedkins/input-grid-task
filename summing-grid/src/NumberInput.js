import React from 'react';

export const NumberInput = ({ name, onChange }) => {
    return (
      <input
        className="grid-border flex-item"
        type="number"
        name={name}
        onChange={onChange}
      />
    );
}

export default NumberInput;
