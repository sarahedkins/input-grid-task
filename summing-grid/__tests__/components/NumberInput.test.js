import React from 'react';
import { shallow } from 'enzyme';
import NumberInput from '../../src/components/NumberInput';

describe('NumberInput', () => {
  it('should render', () => {
    const props = {
      name: "test-component",
      onChange: jest.fn(),
    };
    const wrapper = shallow(<NumberInput {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
