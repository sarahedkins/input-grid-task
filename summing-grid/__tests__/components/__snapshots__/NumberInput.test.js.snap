// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NumberInput
    name="test-component"
    onChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "grid-border flex-item",
      "name": "test-component",
      "onChange": [MockFunction],
      "type": "number",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "grid-border flex-item",
        "name": "test-component",
        "onChange": [MockFunction],
        "type": "number",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
